web/src/app/admin/skills/skills.component.ts

```diff
@@ -1,14 +1,95 @@
-import {Component, OnInit} from '@angular/core';
+import { Component, OnInit } from '@angular/core';
+import { SkillsServiceApp } from './shared/skills.serviceApp';
+import { SkillViewModel } from '../../webapi/models/skill-view-model';
 
 @Component({
   selector: 'app-skills',
   templateUrl: 'skills.component.html'
 })
 
 export class SkillsComponent implements OnInit {
-  constructor() {
-  }
 
+  skillsModel: SkillViewModel = {} as SkillViewModel;
+  skills: SkillViewModel[] = [];
+  isCreate: boolean = false;
+  isUpdate: boolean = false;
+  
+  isSkillExists: boolean = false;
+
+  addskills(skills: SkillViewModel) {
+  }
+  constructor(private SkillsServiceApp: SkillsServiceApp) {
+  }
   ngOnInit() {
+    this.listSkill();
+  }
+  msg: any = "";
+
+  onSubmit(skillsform) {
+    if (skillsform.valid) {
+      if (this.skillsModel['skillId'] === undefined) {
+        this.SkillsServiceApp.addSkill(this.skillsModel).subscribe(
+          (data) => {
+            this.skills.push(data.body);
+            this.msg = " created successfully";
+          }
+        );
+        this.skillsModel.name = "";
+        this.skillsModel.description = "";
+      } else {
+        this.SkillsServiceApp.updateSkill(this.skillsModel).subscribe(
+          (data) => {
+            this.listSkill();
+            this.msg = " updated successfully";
+          }
+        );
+      }
+      if (name == this.skillsModel.name) {
+      }
+      this.skillsModel.name = "";
+      this.skillsModel.description = "";
+    }
+  }
+  checkNameExitsOnBlur() {
+    let $this = this;
+    let skillName = $this.skillsModel.name;
+    this.skills.forEach(function (skill) {
+      if (skill['name'] === skillName) {
+        $this.isSkillExists = true;
+        return;
+      }
```

Comment:
need to check duplicate using api methods

Reply:
done